### PR TITLE
Use header hash as block ID for Protobuf blocks

### LINKED ISSUE #265: [V1.2] Replace BlockSignature with hash digest of block header as Block ID
**Аbstract**
Currently the block signature is used as a block ID to reference to. It should be replaced with hash digest of block's header. 

** Specification**

- [x] Replace block signature with hash digest of block's header
- [x] Add new network messages to request and receive blocks IDs

### PR DESCRIPTION
**Pull request by zer0main**
Closes #265.

### CODE REVIEW

File: pkg/proto/block.go [hunk 1]
```diff
@@ -34,23 +36,133 @@ func (nc *NxtConsensus) BinarySize() int {
...
+)
+
+type BlockID struct {
+	sig    crypto.Signature
+	dig    crypto.Digest
+	idType BlockIDType
+}
+
+func NewBlockIDFromBase58(s string) (BlockID, error) {
+	dig, err := crypto.NewDigestFromBase58(s)
```
Comment (alexeykiselev) [inline pkg/proto/block.go]: Is it correct to try to created a digest, that is shorter, first? It seems like we can build digest from any signature...

Comment (alexeykiselev) [reply pkg/proto/block.go]: May be it is better to convert string to bytes and then use `NewBlockIDFromBytes`?

Comment (zer0main) [reply pkg/proto/block.go]: It checks that decoded length is exactly 32, so it shouldn't accept signatures.

Comment (zer0main) [reply pkg/proto/block.go]: It checks that decoded length is exactly 32, so it shouldn't accept signatures. Internally, it converts string to bytes and checks its length for both `Digest` and `Signature`.

**Pull request merged**